client/src/api/v2/socket.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from 'react';
+import { Runtype, Static } from 'runtypes';
+import io from 'socket.io-client';
+
+let socket: SocketIOClient.Socket;
```

Comment:
Is there a reason why don't we initialise `socket` with `io()`? This way we can make it `const` and `getSocket` could also be a one liner.
Unless `io()` has some sort of socket thing that can fail, thus requiring us to manually reconnect it?

Reply:
@angustrau might know why this was done? Otherwise I agree that would be more straightforward.

Reply:
I think this is to just make sure we only create **one** socket.io connection. I'm not sure if setting it to const would help?

Reply:
Well if it's const there `socket` will always be one socket.io connection. The only difference is that it's initialised on start rather than on first `getSocket` call

Reply:
I'll change it to `const`, easy to change back if we find out there was a reason for it.